src/core/tsi/gts_transport_security.c

```diff
@@ -0,0 +1,35 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/core/tsi/gts_transport_security.h"
+#include <grpc/support/alloc.h>
+
+static gts_shared_resource* gts_resource = NULL;
```

Comment:
If you're going to do this lazily at run-time, then the API here is insufficient, because it provides no mechanism to prevent multiple threads from trying to create the struct at the same time.

A better approach would be to have a static struct here, but leave the channel and CQ members uninitialized.  You can then use a mutex in the struct (which would need to be initialized at gRPC initialization time) to control which thread creates the channel and CQ.